scripts/run_stochastic_process.R

```diff
@@ -25,249 +25,404 @@ do_stochastics_2021 <- function(con, test_run) {
+  dir.create(pre_aggregation_path, showWarnings = FALSE, recursive = TRUE)
+  write.csv(files, output_files, row.names = FALSE)
 
   lines <- Inf
   if (isTRUE(test_run)) {
     lines <- 30
   }
 
-
   stub <- "Andromachi Karachaliou - stochastic-burden.202110gavi-2.MenA_Cambridge-Trotter_"
-  continue_on_error(stone_stochastic_process(
-    con,
-    modelling_group = "Cambridge-Trotter",
-    disease = "MenA",
-    touchstone = "202110gavi-3",
-    scenarios = c("mena-no-vaccination", "mena-campaign-default",
-                  "mena-routine-default", "mena-booster-default",
-                  "mena-campaign-ia2030_target", "mena-routine-ia2030_target"),
-    in_path = file.path(in_path, "Cambridge-Trotter"),
-    files = c(paste0(stub, "no_vaccination_:index.csv.xz"),
-              paste0(stub, "campaign_default_:index.csv.xz"),
-              paste0(stub, "routine_default_:index.csv.xz"),
-              paste0(stub, "booster_:index.csv.xz"),
-              paste0(stub, "campaign-ia2030_target_:index.csv.xz"),
-              paste0(stub, "routine-ia2030_target_:index.csv.xz")),
-    cert = "",
-    index_start = 1,
-    index_end = 26,
-    out_path = out_path,
-    pre_aggregation_path = pre_aggregation_path,
-    log_file = log_file,
-    bypass_cert_check = TRUE,
-    lines = lines))
+  modelling_group = "Cambridge-Trotter"
+  disease = "MenA"
+  touchstone = "202110gavi-3"
+  continue_on_error({
+    paths <- stone_stochastic_process(
+      con,
+      modelling_group = modelling_group,
+      disease = disease,
+      touchstone = touchstone,
+      scenarios = c("mena-no-vaccination", "mena-campaign-default",
+                    "mena-routine-default", "mena-booster-default",
+                    "mena-campaign-ia2030_target", "mena-routine-ia2030_target"),
+      in_path = file.path(in_path, "Cambridge-Trotter"),
+      files = c(paste0(stub, "no_vaccination_:index.csv.xz"),
+                paste0(stub, "campaign_default_:index.csv.xz"),
+                paste0(stub, "routine_default_:index.csv.xz"),
+                paste0(stub, "booster_:index.csv.xz"),
+                paste0(stub, "campaign-ia2030_target_:index.csv.xz"),
+                paste0(stub, "routine-ia2030_target_:index.csv.xz")),
+      cert = "",
+      index_start = 1,
+      index_end = 26,
+      out_path = out_path,
+      pre_aggregation_path = pre_aggregation_path,
+      log_file = log_file,
+      bypass_cert_check = TRUE,
+      lines = lines)
+    files <- data.frame(
+      touchstone = touchstone,
+      modelling_group = modelling_group,
+      disease = disease,
+      files = c(paths$all_u5_cal_file, paths$all_u5_coh_file,
+                paths$all_cal_file, paths$all_coh_file),
+      is_cohort = c(FALSE, TRUE, FALSE, TRUE),
+      is_under5 = c(TRUE, TRUE, FALSE, FALSE)
+    )
+    write.table(files, output_files, sep = ",", append = TRUE,
+                row.names = FALSE, col.names = FALSE)
+  })
```

Comment:
Is this `files <- ` the same for all groups?

If so... could consider
```
files_df <- function(touchstone, modelling_group, disease, paths) {
  data.frame(touchstone = touchstone,
  modelling_group = modelling_group,
  disease = disease,
      files = c(paths$all_u5_cal_file, paths$all_u5_coh_file,
                paths$all_cal_file, paths$all_coh_file),
      is_cohort = c(FALSE, TRUE, FALSE, TRUE),
      is_under5 = c(TRUE, TRUE, FALSE, FALSE)
      }
      
      
      write.table(files_df(touchstone, modelling_group, disease, paths), output_files,  ....
```

Reply:
Thanks yeah, much nicer to put that in a helper function